canonical-web-and-design/maas-ui
# feat(ui): add saga for updating a DNS resource then updating the record

## Done

- Migrated actions saga to TypeScript (badly - added 4 `TSFixMe`s)
- Added a saga `updateDomainRecord` for handling what to do when updating a record that could have its name changed and be either an address record or non-address record

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Do to /MAAS/r/domains and click on a domain
- Add an A record and then open the edit form for it
- Change all the fields and submit
- Check that the `updateDNSResource` action is dispatched first, then when that is complete `updateAddressRecord` is dispatched
- Add a TXT record and then open the edit form for it
- Change all the fields and submit
- Check that the `updateDNSResource` action is dispatched first, then when that is complete `updateDNSData` is dispatched
- Check that all the values update correctly

## Fixes

Fixes #2818
